qiskit_machine_learning/neural_networks/estimator_qnn.py

```diff
@@ -0,0 +1,245 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Estimator quantum neural network class"""
+
+from __future__ import annotations
+
+from copy import copy
+import logging
+from typing import Sequence, Tuple
+
+import numpy as np
+from qiskit.algorithms.gradients import (
+    BaseEstimatorGradient,
+    ParamShiftEstimatorGradient,
+    EstimatorGradientResult,
+)
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator, Estimator, EstimatorResult
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from qiskit_machine_learning.exceptions import QiskitMachineLearningError
+
+from .neural_network import NeuralNetwork
+
+logger = logging.getLogger(__name__)
+
+
+class EstimatorQNN(NeuralNetwork):
+    """A Neural Network implementation based on the Estimator primitive.
+
+    The ``EstimatorQNN`` is a neural network that takes in a parametrized quantum circuit
+    with the combined network's feature map (input parameters) and ansatz (weight parameters)
+    and outputs its measurements for the forward and backward passes.
+
+    Attributes:
```

Comment:
I would suggest adding text like this ahead of Attributes otherwise these appear right after the preceding text 


> The following attributes can be set via the initializer but can also be read and
> updated once the EstimatorQNN object has been constructed.

We have done this in algorithms in Terra eg https://github.com/Qiskit/qiskit-terra/blob/ad7e609d10b02fd98090b0fac132113f329825b6/qiskit/algorithms/time_evolvers/pvqd/pvqd.py#L47-L48